pandas/core/ops/methods.py

```diff
@@ -4,121 +4,930 @@
+
+                # error: "SeriesOps" has no attribute "align"
+                left, right = left.align(  # type: ignore[attr-defined]
+                    right, copy=False
+                )
+
+        return left, right
+
+    def _binop(self, other: SeriesOps, func, level=None, fill_value=None):
+        """
+        Perform generic binary operation with optional fill value.
+
+        Parameters
+        ----------
+        other : Series
+        func : binary operator
+        fill_value : float or object
+            Value to substitute for NA/null values. If both Series are NA in a
+            location, the result will be NA regardless of the passed fill value.
+        level : int or level name, default None
+            Broadcast across a level, matching Index values on the
+            passed MultiIndex level.
+
+        Returns
+        -------
+        Series
+        """
+        if not isinstance(other, SeriesOps):
+            raise AssertionError("Other operand must be Series")
+
+        this = self
+
+        if not self.index.equals(other.index):
+            # error: "SeriesOps" has no attribute "align"
+            this, other = self.align(  # type: ignore[attr-defined]
+                other, level=level, join="outer", copy=False
+            )
+
+        this_vals, other_vals = fill_binop(this._values, other._values, fill_value)
+
+        with np.errstate(all="ignore"):
+            result = func(this_vals, other_vals)
+
+        name = get_op_result_name(self, other)
+        return this._construct_result(result, name)
+
+    def _construct_result(
+        self, result: ArrayLike | tuple[ArrayLike, ArrayLike], name: Hashable
+    ) -> Series | tuple[Series, Series]:
+        """
+        Construct an appropriately-labelled Series from the result of an op.
+
+        Parameters
+        ----------
+        result : ndarray or ExtensionArray
+        name : Label
+
+        Returns
+        -------
+        Series
+            In the case of __divmod__ or __rdivmod__, a 2-tuple of Series.
+        """
+        if isinstance(result, tuple):
+            # produced by divmod or rdivmod
+
+            res1 = self._construct_result(result[0], name=name)
+            res2 = self._construct_result(result[1], name=name)
+
+            # GH#33427 assertions to keep mypy happy
+            assert isinstance(res1, SeriesOps)
+            assert isinstance(res2, SeriesOps)
+            return (res1, res2)
+
+        # TODO: result should always be ArrayLike, but this fails for some
+        #  JSONArray tests
+        dtype = getattr(result, "dtype", None)
+        out = self._constructor(
+            result,
+            index=self.index,
+            dtype=dtype,
+        )
+        out = out.__finalize__(self)
+
+        # Set the result's name after __finalize__ is called because __finalize__
+        #  would set it back to self.name
+        out.name = name  # pyright: ignore[reportGeneralTypeIssues]
```

Comment:
@dr-irv how do i get pyright to stop complaining about this?

```
  /home/runner/work/pandas/pandas/pandas/core/ops/methods.py:785:20 - error: Cannot assign member "name" for type "tuple[Series, Series]"
    Member "name" is unknown (reportGeneralTypeIssues)
```

Reply:
I removed the ignore comment locally and run `pre-commit run -av --hook-stage manual pyright_reportGeneralTypeIssues` but did not get an error

Reply:
Was thinking this could be a pyright bug, but might be difficult to create a reproducible example.  At least when I've seen that kind of error, it's usually been a pyright bug